### PR TITLE
REF: call pandas_dtype up-front in Index.__new__

### PR DESCRIPTION
**Pull request by jbrockmendel**
This leaves us in position to drop in dtype-only is_foo_dtype checks

### CODE REVIEW

**Comment by jreback** [comment]
seems fine, i imagine this has a tiny perf hit, which is likely acceptable, can you do a quick measurement.

**Comment by jbrockmendel** [comment]
> i imagine this has a tiny perf hit

yah, i think this only becomes a win once the dtype-only checks go in

**Review by jorisvandenbossche — APPROVED** [review]

File: pandas/core/indexes/base.py [hunk 1]
```diff
@@ -292,54 +294,59 @@ def __new__(
...
+        if "tz" in kwargs:
+            tz = kwargs.pop("tz")
+            validate_tz_from_dtype(dtype, tz)
+            dtype = tz_to_dtype(tz)
+
         if isinstance(data, ABCPandasArray):
             # ensure users don't accidentally put a PandasArray in an index.
             data = data.to_numpy()
 
+        data_dype = getattr(data, "dtype", None)
```
Comment (jreback) [inline pandas/core/indexes/base.py:308]: can you rename to data_dtype, i find this spelling odd :-<

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/indexes/base.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/indexes/base.py:308]: hah, typo

**Review by jreback — APPROVED** [review]

**Pull request merged**